ONSdigital/response-operations-ui
# Filter by ru

# Motivation and Context
There is a need to be able to filter by ru_ref in secure message screens

# What has changed
Passed around both ru_ref and business_id, because secure message uses business id and the ui mandated ru_ref. If only ru ref was passed around it would mean an extra call to party on each tab change. A new compoenent for ru entry was added in the design system . And the extra params added to Secure message calls. As this was being written Secure message was also changing to replace ru_ref with business_id which also changed some of the returned data , this was integrated also .
Several acceptance tests deleted , and replaced with new unit tests in this repo. 

# How to test?
run tests 
Run acceptance tests . Then modify the db to have all the  same survey_id  by modifying the db locally . Then enter an ru ref in the screen and verify that the filter works as expected.Also verify all secure message functionality manually as this has touched every secure message endpoint .


# Links
https://trello.com/c/Zcr7STCe
